Enhance Project Roadmap

alexchristy opened the pull request:
## Checklist

- [x] I have added a version label to my PR (e.g., patch, minor, major).
- [x] I have tested my changes locally and verified they work as expected.
- [x] I have added relevant tests to cover my changes.
- [x] I have made any necessary updates to the documentation.
- [x] I have made any necessary updates to the CLI.
- [x] I have made any necessary updates to the frontend.

## Description

I moved the roadmap from the _Introduction_ to its own documentation page. It now consists of a table with high level features, with a quick description, a status, and the projected completion date. Then each entry is linked to a GitHub milestone for the project that groups together several issues that contribute to achieving that milestone.

The idea is that users can quickly see our progress on the roadmap, and we as developers can see how our day-to-day work and week-to-week PRs contribute to the overall project vision.

In each milestone, I did my best to break up the work into issues that represent small chunks of work to help keep PRs smaller and more reviewable (looking at #102 :eyes:). The issues that make up a milestone are flexible, and we should add or remove issues from milestone's as requirements evolve, and learning happens.

alexchristy commented on docs/src/roadmap.md:
```diff
@@ -0,0 +1,23 @@
+| [**Live Environment Management**](https://github.com/OpenLabsHQ/OpenLabs/milestone/1) | Dynamically add/remove hosts, power them on/off, and manage firewall rules in a deployed Range without a full redeploy.       | 🏗️ In Progress      | Q3 2025              |
+| [**Lab Snapshots & Cloning**](https://github.com/OpenLabsHQ/OpenLabs/milestone/2) | Save the complete state of a running host or an entire Range. Deploy perfect, pre-configured clones from a snapshot.           | 🗓️ Planned | Q3 2025              |
+| [**Automated Range Cleanup**](https://github.com/OpenLabsHQ/OpenLabs/milestone/3) | Set expiration timers to automatically shut down or destroy deployed Ranges, helping reduce cloud costs. | 🗓️ Planned | Q3 2025 |
+| **Azure Cloud Provider** | Define your lab once and deploy it to Microsoft Azure, in addition to our existing AWS support.                               | 🗓️ Planned      | Q3 2025    |
```
@Adamkadaban not sure if you want to create a milestone for this.

Adamkadaban reviewed (approved):
Looks good. IMO the roadmap is not the most relevant thing to users, so I think it should go near the end of the docs instead of near the beginning.

alexchristy commented:
> Looks good. IMO the roadmap is not the most relevant thing to users, so I think it should go near the end of the docs instead of near the beginning.

I agree most users are going to be looking how to deploy/use the app when they come to the docs. Just moved it the end of the index. Changed here: https://github.com/OpenLabsHQ/OpenLabs/pull/138/commits/d968fa86b8d280fc1cd0b29fc34c3e4487a08382

Nareshp1 reviewed (approved):
LGTM
